air/src/stack/system_ops/mod.rs

```diff
@@ -0,0 +1,134 @@
+    8,
+    // There's one unique constraint in FMPADD system operation in which the top element in the
+    // stack is is incremented by fmp register value.
+    8,
+];
+
+// SYSTEM OPERATIONS TRANSITION CONSTRAINTS
+// ================================================================================================
+
+/// Builds the transition constraint degrees for the system operations.
+pub fn get_transition_constraint_degrees() -> Vec<TransitionConstraintDegree> {
+    CONSTRAINT_DEGREES
+        .iter()
+        .map(|&degree| TransitionConstraintDegree::new(degree))
+        .collect()
+}
+
+/// Returns the number of transition constraints for the system operations.
+pub fn get_transition_constraint_count() -> usize {
+    NUM_CONSTRAINTS
+}
+
+/// Enforces constraints for the system operations.
+pub fn enforce_constraints<E: FieldElement>(
+    frame: &EvaluationFrame<E>,
+    result: &mut [E],
+    op_flag: &OpFlags<E>,
+) -> usize {
+    let mut index = 0;
+
+    // fetching the respective flags of noop and fmpadd from op_flag struct.
+    let noop_op_flag = op_flag.noop();
+    let fmpadd_op_flag = op_flag.fmpadd();
+
+    // Enforce unique constaints of the SWAP operation.
+    index += enforce_noop_unique_constraints(frame, result, noop_op_flag);
+
+    index += enforce_fmpadd_unique_constraints(frame, &mut result[index..], fmpadd_op_flag);
+
+    index
+}
+
+// TRANSITION CONSTRAINT HELPERS
+// ================================================================================================
+
+/// Enforces unique constraints of the NOOP operation. The NOOP operation does nothing to the stack
+/// stack and the entire stack state is copied over in the next cycle.
+/// - Enforces that all the elements in the next cycles are the same as in current cycle.
+pub fn enforce_noop_unique_constraints<E: FieldElement>(
+    frame: &EvaluationFrame<E>,
+    result: &mut [E],
+    op_flag: E,
+) -> usize {
+    let mut constraint_offset = 0;
+
+    // Enforces the first element in the next frame is same as first element in the current frame.
+    result[constraint_offset] = op_flag * are_equal(frame.selector(0), frame.selector_next(0));
+
+    constraint_offset += 1;
+
+    constraint_offset
+}
+
+/// Enforces unique constraints of the FMPADD operation. The FMPADD operation increments the
+/// top element in the stack by fmp value. Therefore, the following constraints are enforced:
+/// - The next element in the next frame should be equal to the addition of frist element in the
+///   current frame and the fmp value.
+pub fn enforce_fmpadd_unique_constraints<E: FieldElement>(
+    frame: &EvaluationFrame<E>,
+    result: &mut [E],
+    op_flag: E,
+) -> usize {
+    let mut constraint_offset = 0;
+
+    // Enforces the first element in the next frame is incremented by fmp value.
+    result[constraint_offset] = op_flag
+        * are_equal(
+            frame.selector(0) + frame.selector_fmp() - frame.selector_next(0),
+            E::ZERO,
```

Comment:
This can be simplified as: `are_equal(frame.selector(0) + frame.selector_fmp(), frame.selector_next(0))`